### PR TITLE
Update to v0.15.0

### PR DESCRIPTION
**Pull request by sigma-andex**
- Use es modules
- Use spago
- Update dependencies

### CODE REVIEW

**Comment by sigma-andex** [comment]
@sharkdp [this test](https://github.com/sharkdp/purescript-bigints/pull/40/files#diff-71732b478b4808898d86c8591ad7ab46d8122c1e4facec4a9151ac49efba905dR108) failed and only passes with the change made. I am not sure whether this correct.

**Comment by sharkdp** [comment]
> @sharkdp [this test](https://github.com/sharkdp/purescript-bigints/pull/40/files#diff-71732b478b4808898d86c8591ad7ab46d8122c1e4facec4a9151ac49efba905dR108) failed and only passes with the change made. I am not sure whether this correct.

well, that is.. unexpected? Are there some upstream changes that would indicate a breaking change?

File: packages.dhall [hunk 1]
```diff
@@ -0,0 +1,6 @@
+let upstream =
+      https://raw.githubusercontent.com/purescript/package-sets/prepare-0.15/src/packages.dhall
+        sha256:0c9c7cd174b7529f2862d52fbfee8cffe1ba8b9cfed3f0f600bdb5be0a14c0ff
```
Comment (JordanMartinez) [inline packages.dhall]: Note: this hash shouldn't be included for the time being as the package set's contents will change until we have a  released version...

**Comment by sigma-andex** [comment]
> > @sharkdp [this test](https://github.com/sharkdp/purescript-bigints/pull/40/files#diff-71732b478b4808898d86c8591ad7ab46d8122c1e4facec4a9151ac49efba905dR108) failed and only passes with the change made. I am not sure whether this correct.
> 
> well, that is.. unexpected? Are there some upstream changes that would indicate a breaking change?

I haven't found anything... not sure what to think about this.

Apart from that, the 0.15 update is now complete

File: package.json [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+    "test": "spago test"
   },
   "dependencies": {
-    "big-integer": "^1.6.27"
+    "big-integer": "^1.6.51"
   },
   "devDependencies": {
     "bower": "^1.8.12",
-    "pulp": "^15.0.0",
+    "pulp": "^16.0.0-0",
```
Comment (JordanMartinez) [inline package.json]: This should be `16.0.0` now.

**Comment by JordanMartinez** [comment]
@sigma-andex Is this PR using `spago@0.20.8` now?

Also, mind adding a `ToString` for `BigInt`, which can reflect type-level `Int`s to a `BigInt`? Something like this should work:
```purs
import Prim.Int (class ToString)
import Data.Reflectable (class Reflectable)

reflectBigInt :: forall i. ToString i sym => Reflectable sym String => Proxy i -> BigInt
reflectBigInt p = fromString $ reflectType p
```

**Comment by sigma-andex** [comment]
> ```purescript
> import Prim.Int (class ToString)
> import Data.Reflectable (class Reflectable)
> ```

Not sure if this is possible, `fromString` returns a `Maybe`

**Comment by JordanMartinez** [comment]
> > ```purescript
> > import Prim.Int (class ToString)
> > import Data.Reflectable (class Reflectable)
> > ```
> 
> Not sure if this is possible, `fromString` returns a `Maybe`

`fromString` has to return `Maybe` because it must work on all `String` types. However, if we can guarantee that the `String` type stores an integer, which it does in this case, then we don't need that `Maybe`.

So, we should actually just use `fromBase` here, but without the `Maybe` wrapper. In other words, exposing FFI to just [this function](https://github.com/sharkdp/purescript-bigints/blob/master/src/Data/BigInt.js#L10=) and using that in the implementation.

**Review by sharkdp — APPROVED** [review]
Thank you!

**Comment by JordanMartinez** [comment]
This PR should now be ready to go.

**Comment by sigma-andex** [comment]
Thx @JordanMartinez

File: src/Data/BigInt.purs [hunk 3]
```diff
@@ -111,6 +115,18 @@ foreign import shl :: BigInt -> Number -> BigInt
 -- | Shift the bits right and maintain pos/neg.
 foreign import shr :: BigInt -> Number -> BigInt
 
+-- Note: this function should not be exported!
+-- It's only safe if used with type-level integers.
+foreign import fromTypeLevelInt :: String -> BigInt
```
Comment (thomashoneyman) [inline src/Data/BigInt.purs:120]: What about renaming this `fromTypeLevelIntImpl` and then having `fromTypeLevelInt` be the exported name? It's not immediately clear to me what `TL` means in the function name.

**Comment by thomashoneyman** [comment]
https://pursuit.purescript.org/packages/purescript-bigints/7.0.1

**Pull request merged**